### PR TITLE
Fix for issue #10.

### PR DESCRIPTION
**Pull request by LiamGoodacre**
https://github.com/natefaubion/adt.js/issues/10

Added tests to replicate the issue.
Fixed issue by binding constructors built with `.type`.

### CODE REVIEW

**Comment by LiamGoodacre** [comment]
Oh I forgot about generating the minified file - how do I do this?

**Comment by natefaubion** [comment]
I'll minify it, don't worry about it.

**Comment by LiamGoodacre** [comment]
Amended with semicolons :]

**Comment by natefaubion** [comment]
Thanks!

File: test/core.js [hunk 1]
```diff
@@ -236,4 +236,67 @@ suite('Core', function () {
     ok('Multiple types', mult('foo') && mult(42) && mult(null) === null);
     throws('Multiple throws', function () { mult(true); }, TypeError);
   });
+
+  suite('Calling as method from another other instance', function () {
```
Comment (michaelficarra) [inline test/core.js:240]: "another other"?

Comment (LiamGoodacre) [inline test/core.js:240]: Oops, maybe rewrite it as `Calling as a method on another instance:`?

**Pull request merged**